components/sidebar/sidebar_channel/sidebar_channel.tsx

```diff
@@ -82,18 +82,44 @@ type Props = {
 };
 
 type State = {
-
+    show: boolean;
 };
 
 export default class SidebarChannel extends React.PureComponent<Props, State> {
     static defaultProps = {
         isDraggable: true,
     }
 
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            show: this.isCollapsed(props),
+        };
+    }
+
     isCollapsed = (props: Props) => {
         return props.isCategoryDragged || (props.isCategoryCollapsed && !this.props.isUnread && !props.isCurrentChannel);
     }
 
+    componentDidMount() {
+        const channelElement = this.getRef();
+        if (channelElement) {
+            channelElement.addEventListener('transitionend', this.handleTransitionEnd);
```

Comment:
Is there some reason that we can't add this listener by passing an `onTransitionEnd` prop to the component?

Reply:
✅ I removed this event listener and added the handler directly to the element